src/Microsoft.DotNet.RemoteExecutor/src/RemoteExecutor.cs

```diff
@@ -51,17 +51,30 @@ static RemoteExecutor()
                 return;
             }
 
-            HostRunnerName = System.IO.Path.GetFileName(processFileName);
             Path = typeof(RemoteExecutor).Assembly.Location;
 
             if (IsNetCore())
             {
                 HostRunner = processFileName;
+
+                // Running with an apphost, eg. Visual Studio testhost. We should attempt to find and use dotnet.exe.
+                // This is a Windows-only workaround.
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && !HostRunner.EndsWith("dotnet.exe", StringComparison.OrdinalIgnoreCase))
+                {
+                    string runtimePath = System.IO.Path.GetDirectoryName(typeof(object).Assembly.Location);
+                    string dotnetExe = System.IO.Path.Combine(runtimePath, "..", "..", "..", "dotnet.exe");
```

Comment:
I personally would prefer something like:
```suggestion
                    string dotnetExe = Path.Combine(Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(runtimePath))),"dotnet.exe");
                    The usage of relative path segments may confuse other parts of the system eventually.
```

It would need to be a bit more complicated as it needs to check for empty result of `GetDirectoryName`.